webserver/src/main/java/io/kestra/webserver/services/ai/gemini/GeminiAiService.java

```diff
@@ -0,0 +1,362 @@
+            """.formatted(
+            serializedPlugins
+        );
+
+
+        return LlmAgent.builder()
+            .name("kestra_plugin_relevance_agent")
+            .description("An agent to build a Kestra Flow YAML.")
+            .model(model)
+            .generateContentConfig(GenerateContentConfig.builder().temperature(0.9f).maxOutputTokens(1024).build())
+            .instruction(pluginRelevanceInstructions)
+            .outputKey("mostRelevantTypes")
+            .afterAgentCallback(callbackContext -> {
+                String mostRelevantTypes = ((String) callbackContext.state().get("mostRelevantTypes")).trim();
+                if (POSSIBLE_ERROR_MESSAGES.contains(mostRelevantTypes)) {
+                    callbackContext.state().put("responseForUser", mostRelevantTypes);
+                    return shortCircuitLlmResponse(callbackContext, mostRelevantTypes);
+                }
+
+                if (mostRelevantTypes.isEmpty()) {
+                    return Maybe.empty();
+                }
+
+                String[] identifiedTypes = mostRelevantTypes.split(" ?, ?");
+
+                JsonNode minifiedSchema = JacksonMapper.ofJson().convertValue(jsonSchemaGenerator.schemas(Flow.class, false, Optional.of(Arrays.asList(identifiedTypes))), JsonNode.class);
+                minifySchema(minifiedSchema);
+                try {
+                    PluginRegistry registry = pluginRegistry;
+                    callbackContext.state().put(
+                        "flowSchema",
+                        JacksonMapper.ofJson().writeValueAsString(minifiedSchema)
+                    );
+                } catch (JsonProcessingException e) {
+                    log.error("Failed to serialize flow schema for AI Agent", e);
+                }
+
+                return Maybe.empty();
+            })
+            .build();
+    }
+
+    private static LlmAgent createYamlBuilderAgent(Gemini model) {
+        return LlmAgent.builder()
+            .name("kestra_init_flow_builder_agent")
+            .description("An agent to bootstrap a Kestra Flow YAML.")
+            .model(model)
+            .generateContentConfig(GenerateContentConfig.builder().temperature(0.9f).maxOutputTokens(50000).build())
+            .outputKey("responseForUser")
+            .beforeAgentCallback((beforeAgentCallback) -> {
+                Object responseForUser = beforeAgentCallback.state().get("responseForUser");
+                if (responseForUser != null) {
+                    return shortCircuitLlmResponse(beforeAgentCallback, responseForUser.toString());
+                }
+
+                return Maybe.empty();
+            })
+            .instruction(
+                """
+                    You are part of a Kestra Flow builder service. You are an expert in generating Kestra Flow YAML. Your task is to generate a valid Kestra Flow YAML based strictly on the following schema:
```

Comment:
```suggestion
                    You are an expert in generating Kestra Flow YAML. Your task is to generate a valid Kestra Flow YAML strictly following the schema below:
```

afaik the Kestra Flow builder service is not needed, it won't affect the result